cmd/create/cluster/cmd.go

```diff
@@ -587,49 +587,93 @@ func run(cmd *cobra.Command, _ []string) {
 			}
-			selectedARN := ""
-			if len(roleARNs) > 1 {
-				reporter.Warnf("More than one %s role found", role.Name)
-				selectedARN, err = interactive.GetOption(interactive.Input{
-					Question: fmt.Sprintf("%s role ARN", role.Name),
-					Help:     cmd.Flags().Lookup(role.Flag).Usage,
-					Options:  roleARNs,
-					Default:  roleARNs[0],
-					Required: true,
-				})
+		} else if len(roleARNs) == 1 {
+			if !output.HasFlag() || reporter.IsTerminal() {
+				reporter.Infof("Using %s for the %s role", roleARNs[0], role.Name)
+			}
+			roleARN = roleARNs[0]
+		} else {
+			reporter.Warnf("No account roles found. You will need to manually set them in the " +
+				"next steps or run 'rosa create account-roles' to create them first.")
+			interactive.Enable()
+		}
+
+		if roleARN != "" {
+			// Get role prefix
+			rolePrefix, err := getAccountRolePrefix(roleARN, role)
+			if err != nil {
+				reporter.Errorf("Failed to find prefix from %s account role", role.Name)
+				os.Exit(1)
+			}
+			reporter.Debugf("Using '%s' as the role prefix", rolePrefix)
+
+			for roleType, role := range aws.AccountRoles {
+				if roleType == aws.InstallerAccountRole {
+					// Already dealt with
+					continue
+				}
+				roleARNs, err := awsClient.FindRoleARNs(roleType, minor)
 				if err != nil {
-					reporter.Errorf("Expected a valid role ARN: %s", err)
+					reporter.Errorf("Failed to find %s role: %s", role.Name, err)
 					os.Exit(1)
 				}
-			} else if len(roleARNs) == 1 {
+				selectedARN := ""
+				for _, rARN := range roleARNs {
+					if strings.Contains(rARN, fmt.Sprintf("%s-%s-Role", rolePrefix, role.Name)) {
+						selectedARN = rARN
+					}
+				}
+				if selectedARN == "" {
+					reporter.Warnf("No %s account roles found. You will need to manually set "+
+						"them in the next steps or run 'rosa create account-roles' to create "+
+						"them first.", role.Name)
+					interactive.Enable()
+					break
+				}
 				if !output.HasFlag() || reporter.IsTerminal() {
-					reporter.Infof("Using %s for the %s role", roleARNs[0], role.Name)
+					reporter.Infof("Using %s for the %s role", selectedARN, role.Name)
 				}
-				selectedARN = roleARNs[0]
-			} else {
-				reporter.Warnf("No account roles found. You will need to manually set them in the " +
-					"next steps or run 'rosa create account-roles' to create them first.")
-				interactive.Enable()
-				break
-			}
-			switch roleType {
-			case "installer":
-				roleARN = selectedARN
-			case "support":
-				supportRoleARN = selectedARN
-			case "instance_controlplane":
-				controlPlaneRoleARN = selectedARN
-			case "instance_worker":
-				workerRoleARN = selectedARN
+				switch roleType {
+				case aws.InstallerAccountRole:
+					roleARN = selectedARN
+				case aws.SupportAccountRole:
+					supportRoleARN = selectedARN
+				case aws.ControlPlaneAccountRole:
+					controlPlaneRoleARN = selectedARN
+				case aws.WorkerAccountRole:
+					workerRoleARN = selectedARN
+				}
+				hasRoles = true
```

Comment:
We run into an issue here I think, if one of the roles does not have the prefix, hasRoles can still end up true if the last role in the list has the prefix. Which we will then skip the manual prompt to enter the role. 
Example : https://gist.github.com/ciaranRoche/53c19f395688f021048b4aa25ef68564#file-when_on_role_does_not_exist

Maybe we can drop the `hasRoles` flag and explicitly check if the `role == ""` when determining to prompt the user input for the role. Will also need to add a validator to those prompts to ensure the correct prefix.